server/src/main/java/org/opensearch/cluster/metadata/MetadataUpdateSettingsService.java

```diff
@@ -260,6 +264,34 @@ public ClusterState execute(ClusterState currentState) {
                         }
                     }
 
+                    if (IndexMetadata.INDEX_NUMBER_OF_SEARCH_REPLICAS_SETTING.exists(openSettings)) {
+                        if (FeatureFlags.isEnabled(FeatureFlags.READER_WRITER_SPLIT_EXPERIMENTAL_SETTING)) {
+                            validateSearchReplicaCountSettings(normalizedSettings, request.indices(), currentState);
+                        }
+                        final int updatedNumberOfSearchReplicas = IndexMetadata.INDEX_NUMBER_OF_SEARCH_REPLICAS_SETTING.get(openSettings);
+                        if (preserveExisting == false) {
+                            // TODO: Maybe honor awareness validation to search replicas?
```

Comment:
Can we answer these TODO questions before committing this change?

Reply:
updated the todo